infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -462,10 +460,29 @@ protected void initChannel(@Nonnull Channel ch) throws Exception {
                     SSLEngine engine = sslContext.newEngine(ch.alloc());
                     engine.setEnabledCipherSuites(enabledTlsCipherSuites);
                     engine.setEnabledProtocols(enabledTlsProtocols);
+
                     if (tlsMutualAuthEnabled) {
                         engine.setNeedClientAuth(true);
                     }
-                    ch.pipeline().addLast("ssl", new SslHandler(engine));
+
+                    SslHandler sslHandler = new SslHandler(engine);
+
+                    // Add listener to handle SSL Handshake Exceptions in order to properly log
+                    // the target IP failing to authenticate against this server.
+                    sslHandler.handshakeFuture().addListener(channelFuture -> {
+                        if (channelFuture.cause() != null && channelFuture.cause().getClass()
+                                .equals(SSLHandshakeException.class)) {
+                            String peerHost;
```

Comment:
Can't you just use 
`ch.remoteAddress()`
instead of casting?

Example:
https://github.com/CorfuDB/CorfuDB/blob/master/runtime/src/main/java/org/corfudb/protocols/wireprotocol/ClientHandshakeHandler.java#L153

Reply:
Done. It also prints the port info but no harm in this.

Reply:
On re-connection the client will bind to a new port, that's how I was able to tell that the client was spinning on reconnecting (the ports kept changing on every request)